### PR TITLE
Make besselroots type stable

### PR DESCRIPTION
**Pull request by daanhb**
I was chasing memory allocations when reviewing Peter Opsomer's fast Laguerre expansions code and noted besselroots is not type stable. The culprit I think is returning [] when n is 0. The code now initializes x as zeros(n).

In the process I removed all dots from McMahon (the previous code was vectorized, and the same result can be achieved still using McMahon.(nu, 1:n), i.e., using broadcast) and I moved some constants outside of function definitions. This again reduces memory allocations ever-so-slightly.

Slight gains may still be made by writing a besselroots!(x, nu, n) function, with x preallocated.

### CODE REVIEW

**Comment by daanhb** [comment]
Apparently the tests with Julia nightly failed somewhere when testing Laguerre, but with 0.6 all tests pass. In any case, the Laguerre code is changed completely in the simple_expansions branch (using Peter's explicit expansions). Peter has an open pull request for that, we'll happily fix any remaining problems.

**Comment by dlfivefifty** [comment]
We shouldn't worry too much about nightly at the moment.

File: src/besselroots.jl [hunk 1]
```diff
@@ -13,77 +13,74 @@ function besselroots(nu::Float64, n::Integer)
...
-          mu*(-287149133 .+5592657 .*mu))))) / 10463949619200.
-    a13 = 576 .*(423748443625564327. + mu*(-100847472093088506 .+mu*(8929489333108377. +
-        mu*(-426353946885548 .+mu*(13172003634537 .+mu*(-291245357370. + mu*4148944183.)))))) / 13059009124761600.
-    for k=1:n b[k] = .25*(2*nu+4*k-1)*pi end # beta
+    mu = 4nu^2
+    a1 = 1 / 8
+    a3 = (7mu-31) / 384
+    a5 = 4*(3779+mu*(-982+83mu)) / 61440 # Evaluate via Horner's method.
+    a7 = 6*(-6277237+mu*(1585743+mu*(-153855+6949mu))) / 20643840
+    a9 = 144*(2092163573+mu*(-512062548.+mu*(48010494+mu*(-2479316+70197mu)))) / 11890851840
```
Comment (dlfivefifty) [inline src/besselroots.jl]: One left over period

**Comment by dlfivefifty** [comment]
I found one left over period, mind deleting it?

**Comment by daanhb** [comment]
And it's gone.

**Pull request merged**